python/foglamp/plugins/south/systeminfo/systeminfo.py

```diff
@@ -0,0 +1,358 @@
+
+        for interface_name, interface_ip in network_interfaces:
+            network_traffic.update({
+                interface_name: {
+                    "IP": interface_ip,
+                    "networkPacketsReceived": network_calc[interface_name]["bytes_recd_after"] -
+                                              network_calc[interface_name]["bytes_recd_before"],
+                    "networkPacketsSent": network_calc[interface_name]["bytes_sent_after"] -
+                                          network_calc[interface_name]["bytes_sent_before"],
+                }
+            })
+        return network_traffic
+
+    def get_subprocess_result(cmd):
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE).stdout.readlines()
+        # Since "a" contains return value in bytes, convert it to string
+        c = [str(b, 'utf-8').replace('\n', '') for b in a]
+        return c
+
+    def get_system_info():
+        data = {}
+        hostname = get_subprocess_result(cmd='hostname')[0]
+        data.update({
+            "hostName": hostname
+        })
+
+        # Generic search a list for an item starting with the given string. Return item stripped of
+        # the search string if success else return None
+        list_search = lambda x, xlist: next((y[len(x):] for y in xlist if y.startswith(x)), None)
+
+        # Get uptime, load, tasks, cpu usage, memory, swap memory information
+        c2 = get_subprocess_result(cmd='top -n1 -b')[:5]
+        top_line = list_search("top - ", c2).split(',')
+        load_average = {
+                "uptime": top_line[0]+', '+top_line[1],
+                "user(s)": int((top_line[2].strip().split())[0]),
+                "loadAverageOverLast1min": float((top_line[3])[-5:].strip()),
+                "loadAverageOverLast5min": float(top_line[4]),
+                "loadAverageOverLast15min": float(top_line[5])
+        }
+
+        tasks_line = list_search("Tasks: ", c2).split(',')
+        tasks = {}
+        for task in tasks_line:
+            t_item = task.split()
+            tasks.update({t_item[1]: int(t_item[0])})
+
+        cpus_line = list_search("%Cpu(s): ", c2).split(',')
+        cpus = {}
+        for cpu in cpus_line:
+            c_item = cpu.split()
+            cpus.update({c_item[1]: float(c_item[0])})
+
+        memories_line = list_search("KiB Mem : ", c2).split(',')
+        memories = {}
+        for memory in memories_line:
+            m_item = memory.split()
+            memories.update({m_item[1]: int(m_item[0])})
+
+        swaps_line = list_search("KiB Swap: ", c2).replace('.', ',').split(',')
+        swaps = {}
+        for swap in swaps_line:
+            s_item = swap.split()
+            swaps.update({s_item[1]: int(s_item[0])})
+
+        data.update({
+            "loadAverage": load_average,
+            "tasksRunning": tasks,
+            "cpuUsage": cpus,
+            "memoryUsage(KB)": memories,
+            "swapMemory(KB)": swaps
```

Comment:
should we convert to bytes *1024 and just use keys `memoryUsage` and `swapMemory`